### PR TITLE
enhance: Remove compaction plans on the datanode

### PR DESCRIPTION
**Pull request by xiaocai2333**
issue: #33546

### CODE REVIEW

File: internal/datacoord/compaction_task_mix.go [hunk 1]
```diff
@@ -77,10 +75,15 @@ func (t *mixCompactionTask) processExecuting() bool {
...
 	case commonpb.CompactionState_Completed:
 		t.result = result
-		result := t.result
 		if len(result.GetSegments()) == 0 || len(result.GetSegments()) > 1 {
 			log.Info("illegal compaction results")
 			err := t.updateAndSaveTaskMeta(setState(datapb.CompactionTaskState_failed))
+			if err != nil {
+				return false
+			}
+			err = t.sessions.DropCompactionPlan(t.GetNodeID(), &datapb.DropCompactionPlanRequest{
```
Comment (czs007) [inline internal/datacoord/compaction_task_mix.go:84]: move to processFailed?

File: internal/datacoord/session_manager.go [hunk 2]
```diff
@@ -547,6 +548,39 @@ func (c *SessionManagerImpl) QuerySlot(nodeID int64) (*datapb.QuerySlotResponse,
...
+	log := log.With(
+		zap.Int64("nodeID", nodeID),
+		zap.Int64("planID", req.GetPlanID()),
+	)
+	ctx, cancel := context.WithTimeout(context.Background(), Params.DataCoordCfg.CompactionRPCTimeout.GetAsDuration(time.Second))
+	defer cancel()
+	cli, err := c.getClient(ctx, nodeID)
+	if err != nil {
+		log.Warn("failed to get client", zap.Error(err))
+		return err
```
Comment (czs007) [inline internal/datacoord/session_manager.go:561]: return  ErrNodeNotFound?

**Comment by bigsheeper** [comment]
related to https://github.com/milvus-io/milvus/issues/32809

**Pull request merged**